apps/emqx_ldap/src/emqx_ldap.erl

```diff
@@ -273,6 +274,22 @@ on_query(
             Error
     end.
 
+search(Pid, SearchOptions) ->
+    case eldap:search(Pid, SearchOptions) of
+        {error, ldap_closed} ->
```

Comment:
how do you find these? doc only says `{error. term()}`.
what about just restart if error?

Reply:
found it from the source code.
we only want to reconnect when transport (socket) error happens.
other errors are like encode error, decode error so there is no need to reconnect.